### PR TITLE
feat: search backend

### PR DESCRIPTION
**Pull request by avsolatorio**

### CODE REVIEW

File: api/app/document.py [hunk 1]
```diff
@@ -7,23 +8,103 @@
...
+                            if self.fields:
+                                shoulds.append(Q("multi_match", fields=self.fields, query=quote, type="phrase", boost=5))
+                            else:
+                                shoulds.append(Q("multi_match", query=quote, type="phrase", boost=5))
+
+                query = Q('bool', should=shoulds, minimum_should_match=1)
+
+            search = search.query(query)
+
+            print(search.to_dict())
```
Comment (rgbongocan) [inline api/app/document.py:108]: nit: rm print or set level to logging.debug if it's useful (although I haven't setup logging properly yet on our backend outside of prints)

File: api/scripts/qdrant_index_datasets.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import os
+import sys
+
+from app.search.embedding import get_embedding_model, get_qdrant_kwargs
+from app.models import Dataset
+from sqlalchemy import create_engine, select
+from sqlalchemy.orm import sessionmaker
+
+# # TODO: Change this later to the actual DB...
+# DATABASE_CONNECTION = "postgresql://postgres:pass@localhost:5432/public.datasets"
```
Comment (rgbongocan) [inline api/scripts/qdrant_index_datasets.py:14]: rm comments?

File: secrets/embedding.env.tpl [hunk 3]
```diff
@@ -0,0 +1,3 @@
+EMBEDDING_MODEL=mixedbread-ai/mxbai-embed-large-v1
+EMBEDDING_QUERY_PREFIX="Represent this sentence for searching relevant passages: "
```
Comment (rgbongocan) [inline secrets/embedding.env.tpl:2]: What does this do? Just curious!

Comment (avsolatorio) [reply secrets/embedding.env.tpl:2]: Oh, this is the retrieval instruction for the embedding model. See: https://huggingface.co/mixedbread-ai/mxbai-embed-large-v1. Some models claim to perform better when an instruction used during the model training is used.

File: api/app/document.py (hunk 1 above)
Comment (avsolatorio) [reply api/app/document.py]: Rm-ed!

File: api/scripts/qdrant_index_datasets.py (hunk 2 above)
Comment (avsolatorio) [reply api/scripts/qdrant_index_datasets.py]: Rm-ed!

**Pull request merged**